### ISSUE TITLE
Ability to manually push metadata updates

### ISSUE THREAD

**Issue by daredoes** [postlund/pyatv#2003]
### What feature would you like?

The ability to push a metadata update through the `atv` object received from `pyatv.connect` as part of supporting streamed files.

### Describe the solution you'd like

Something like `atv.metadata.update(metadata: AudioMetadata)`

### Any other information to share?

Probably should go into `stream_asyncio` branch

**Comment by postlund** [comment]
Yes, this is something I actually thought of when implementing streaming. I also thought I wrote an issue about it, but I probably just rambled in an related issue. The idea would be to include metadata when streaming a file, e.g.

```python
await atv.stream.stream_file("some_file.mp3", metadata=AudioMetadata(artist="foobar"))
```

**Comment by postlund** [comment]
I decided to merge #2011, so you should be able to do this from the `master` branch. You have some documentation here:

https://pyatv.dev/development/stream/#custom-metadata
